### PR TITLE
YCQL - Cassandra 3.4.2 feature parity

### PR DESCRIPTION
**Pull request by premkumr**
- Lists features that are missing or slightly different from Cassandra 3.4.2

@netlify /preview/api/ycql/cassandra-compatibility

### CODE REVIEW

**Review by ddhodge — COMMENTED** [review]
minor comments

**Comment by premkumr** [comment]
> LGTM. My only issue is that it doesn't show any PROs, only CONs.

@ddorian  - The page is structured to show the critical differences between C*3.4.2 and YCQL. Let me share the original sheet with you & please add features that YCQL has and C*3.4.2 does not, I'll add it in here

File: docs/content/preview/api/ycql/cassandra-compatibility.md [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+## Overview
+
+Yugabyte Cloud Query Language (YCQL) has its roots in the [Cassandra Query Language (CQL)](http://cassandra.apache.org/doc/latest/cql/index.html). This page highlights the important feature support differences between YCQL and Cassandra 3.4.2.
+
+{{<warning>}}
+This page does **not** list all the features supported in YCQL.
+{{</warning>}}
+
+## DDL statements
+|                    |                Table                |                                          |
```
[the remark this answers is not in the document]
Comment (premkumr) [reply docs/content/preview/api/ycql/cassandra-compatibility.md:26]: If we align to left, the third column is not aligned across the different tables. Looks a tad weird. We can discuss this offline.

File: docs/content/preview/api/ycql/_index.md [hunk 2]
```diff
@@ -144,8 +138,7 @@ Primitive Type | Allowed in Key | Type Parameters | Description |
 
 [User-defined data types](ddl_create_type) are also supported.
 
-The following table lists all YCQL primitive types that are not yet implemented.
+## Learn More
```
Comment (ddhodge) [inline docs/content/preview/api/ycql/_index.md:141]:
```suggestion
## Learn more
```

File: docs/content/preview/api/ycql/cassandra-compatibility.md [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+| {{<icon/no>}}  | INSERT _JSON_           |
+{.sno-1}
+
+|                    |              Feature              |                                          |
+| :----------------: | --------------------------------- | ---------------------------------------: |
+|   {{<icon/no>}}    | Materialized Views                |                                          |
+| {{<icon/partial>}} | BATCH                             | Only programattically via BatchStatement |
+{.sno-1}
+
+## Unsupported Datatypes
```
Comment (ddhodge) [inline docs/content/preview/api/ycql/cassandra-compatibility.md:95]:
```suggestion
## Unsupported datatypes
```

File: docs/content/preview/api/ycql/cassandra-compatibility.md [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+
+|               | Type  |
+| :-----------: | ----- |
+| {{<icon/no>}} | Tuple |
+{.sno-1}
+
+## Unsupported Commands
```
Comment (ddhodge) [inline docs/content/preview/api/ycql/cassandra-compatibility.md:102]:
```suggestion
## Unsupported commands
```

File: docs/content/preview/api/ycql/cassandra-compatibility.md [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+| :-----------: | --------------------------------------------------- | ------------------------------------------: |
+| {{<icon/no>}} | [CREATE &vert; DROP] FUNCTION                       |  No support for User Defined Function (UDF) |
+| {{<icon/no>}} | [CREATE &vert; DROP] AGGREGATE                      | No support for User Defined Aggregates(UDA) |
+| {{<icon/no>}} | [CREATE &vert; DROP] TRIGGER                        |                     No support for Triggers |
+| {{<icon/no>}} | [CREATE &vert; DROP &vert; ALTER  &vert; LIST] USER |                    Legacy Cassandra feature |
+| {{<icon/no>}} | LIST ROLES                                          |                                             |
+| {{<icon/no>}} | LIST PERMISSION                                     |                                             |
+{.sno-1}
+
+## Learn More
```
Comment (ddhodge) [inline docs/content/preview/api/ycql/cassandra-compatibility.md:114]:
```suggestion
## Learn more
```

File: docs/content/preview/faq/comparisons/cassandra.md [hunk 6]
```diff
@@ -92,3 +92,8 @@ A few blog posts that highlight how YugabyteDB differs from Apache Cassandra are
 - [DynamoDB vs MongoDB vs Cassandra for Fast Growing Geo-Distributed Apps](https://www.yugabyte.com/blog/dynamodb-vs-mongodb-vs-cassandra-for-fast-growing-geo-distributed-apps/)
 - [YugabyteDB 1.1 New Feature: Document Data Modeling with the JSON Data Type](https://www.yugabyte.com/blog/yugabyte-db-1-1-new-feature-document-data-modeling-with-json-data-type/)
 - [YugabyteDB 1.1 New Feature: Speeding Up Queries with Secondary Indexes](https://www.yugabyte.com/blog/yugabyte-db-1-1-new-feature-speeding-up-queries-with-secondary-indexes/)
+
+## Learn More
```
Comment (ddhodge) [inline docs/content/preview/faq/comparisons/cassandra.md:96]:
```suggestion
## Learn more
```

File: docs/content/preview/faq/comparisons/cassandra.md [hunk 7]
```diff
@@ -92,3 +92,8 @@ A few blog posts that highlight how YugabyteDB differs from Apache Cassandra are
...
+
+- [YCQL command reference](../../../api/ycql/)
```
Comment (ddhodge) [inline docs/content/preview/faq/comparisons/cassandra.md:98]:
```suggestion
- [YCQL API](../../../api/ycql/)
```

File: docs/content/preview/api/ycql/_index.md [hunk 8]
```diff
@@ -5,7 +5,7 @@ linkTitle: YCQL
 description: YCQL is a semi-relational API that is best fit for internet-scale OLTP & HTAP applications.
 summary: Reference for the YCQL API
 image: /images/section_icons/api/ycql.png
-headcontent:
+headcontent: Cassandra compatible API
```
Comment (ddhodge) [inline docs/content/preview/api/ycql/_index.md:8]:
```suggestion
headcontent: Cassandra-compatible API
```

File: docs/content/preview/api/ycql/_index.md [hunk 9]
```diff
@@ -144,8 +138,7 @@ Primitive Type | Allowed in Key | Type Parameters | Description |
...
 [User-defined data types](ddl_create_type) are also supported.
 
-The following table lists all YCQL primitive types that are not yet implemented.
+## Learn More
 
-Primitive Type |
----------------|
-`TUPLE` |
+- [Advantages of YCQL over Cassandra](../../faq/comparisons/cassandra)
+- [Differences in features over Cassandra](./cassandra-compatibility)
```
Comment (ddhodge) [inline docs/content/preview/api/ycql/_index.md:144]:
```suggestion
- [YCQL - Cassandra 3.4 compatibility](./cassandra-compatibility)
```

File: docs/content/preview/api/ycql/cassandra-compatibility.md [hunk 10]
```diff
@@ -0,0 +1,117 @@
...
+    parent: api-cassandra
+    weight: 1190
+type: docs
+---
+
+## Overview
+
+Yugabyte Cloud Query Language (YCQL) has its roots in the [Cassandra Query Language (CQL)](http://cassandra.apache.org/doc/latest/cql/index.html). This page highlights the important feature support differences between YCQL and Cassandra 3.4.2.
+
+{{<warning>}}
```
Comment (ddhodge) [inline docs/content/preview/api/ycql/cassandra-compatibility.md:20]:
```suggestion
{{<note>}}
```

File: docs/content/preview/api/ycql/cassandra-compatibility.md [hunk 11]
```diff
@@ -0,0 +1,117 @@
...
+This page does **not** list all the features supported in YCQL. For all the features and commands supported, see [YCQL Reference](../../ycql). For advantages of YCQL over Cassandra, see [Comparison with Apache Cassandra](../../../faq/comparisons/cassandra)
+{{</warning>}}
```
Comment (ddhodge) [inline docs/content/preview/api/ycql/cassandra-compatibility.md:22]:
```suggestion
{{</note>}}
```

File: docs/content/preview/api/ycql/cassandra-compatibility.md [hunk 12]
```diff
@@ -0,0 +1,117 @@
...
+    weight: 1190
+type: docs
+---
+
+## Overview
+
+Yugabyte Cloud Query Language (YCQL) has its roots in the [Cassandra Query Language (CQL)](http://cassandra.apache.org/doc/latest/cql/index.html). This page highlights the important feature support differences between YCQL and Cassandra 3.4.2.
+
+{{<warning>}}
+This page does **not** list all the features supported in YCQL. For all the features and commands supported, see [YCQL Reference](../../ycql). For advantages of YCQL over Cassandra, see [Comparison with Apache Cassandra](../../../faq/comparisons/cassandra)
```
Comment (ddhodge) [inline docs/content/preview/api/ycql/cassandra-compatibility.md:21]:
```suggestion
This page does **not** list all the features supported in YCQL. For all the features and commands supported, see [YCQL Reference](../../ycql). For advantages of YCQL over Cassandra, see [Comparison with Apache Cassandra](../../../faq/comparisons/cassandra).
```

File: docs/content/preview/api/ycql/cassandra-compatibility.md [hunk 13]
```diff
@@ -0,0 +1,117 @@
...
+| {{<icon/no>}} | [CREATE &vert; DROP] AGGREGATE                      | No support for User Defined Aggregates(UDA) |
+| {{<icon/no>}} | [CREATE &vert; DROP] TRIGGER                        |                     No support for Triggers |
+| {{<icon/no>}} | [CREATE &vert; DROP &vert; ALTER  &vert; LIST] USER |                    Legacy Cassandra feature |
+| {{<icon/no>}} | LIST ROLES                                          |                                             |
+| {{<icon/no>}} | LIST PERMISSION                                     |                                             |
+{.sno-1}
+
+## Learn More
+
+- [Advantages of YCQL over Cassandra](../../../faq/comparisons/cassandra)
```
Comment (ddhodge) [inline docs/content/preview/api/ycql/cassandra-compatibility.md:116]:
```suggestion
- [Comparison with Apache Cassandra](../../../faq/comparisons/cassandra)
```

File: docs/content/preview/api/ycql/cassandra-compatibility.md [hunk 14]
```diff
@@ -0,0 +1,117 @@
...
+menu:
+  preview:
+    parent: api-cassandra
+    weight: 1190
+type: docs
+---
+
+## Overview
+
+Yugabyte Cloud Query Language (YCQL) has its roots in the [Cassandra Query Language (CQL)](http://cassandra.apache.org/doc/latest/cql/index.html). This page highlights the important feature support differences between YCQL and Cassandra 3.4.2.
```
Comment (ddhodge) [inline docs/content/preview/api/ycql/cassandra-compatibility.md:18]:
```suggestion
Yugabyte Cloud Query Language (YCQL) has its roots in the [Cassandra Query Language (CQL)](http://cassandra.apache.org/doc/latest/cql/index.html). This page highlights the important differences in feature support between YCQL and Cassandra 3.4.2.
```

**Review by ddhodge — APPROVED** [review]

**Pull request merged**